src/Quartz/Util/StringKeyDirtyFlagMap.cs

```diff
@@ -548,6 +604,23 @@ public virtual bool TryGetNullableGuid(string key, out Guid? value)
             }
         }
 
+        /// <summary>
+        /// Try to retrieve the identified <see cref="string" /> value from the <see cref="JobDataMap" />.
+        /// </summary>
+        public virtual bool TryGetNullableString(string key, out string? value)
```

Comment:
I think a better name would be `TryGetString` as string is already nullable, the Guid overload with `Nullable` is a bit of exception to the rule I think, thoughts?

Reply:
You make a good point, I was trying to conform to the existing pattern.  Will rename to `TryGetString`..

Reply:
I should remove  this from GetString given then that null is valid .
``` c#
 if (obj is null)
            {
                throw new InvalidCastException("Identified object is not a String.");
            }
```